basics/Main.java

```diff
@@ -23,29 +24,38 @@ public static String pluralize (String word, int number) {
         String pluralWord = "";
         if(number > 1) {
             pluralWord = word + "s";
-        } else pluralWord = word;
-
+        }else if (number == 0) {
+            pluralWord = "";
+        } else {
+            pluralWord = word;
+        }
         return pluralWord;
     }
 
     //flipping coin method
     public static String flipNHeads (int flipCount) {
         int flipCounter = 0;
-        int headsCounter = 0;
+        int headsCounter = 1;
         String flipResult = "";
+        String flipSummary = "";
         while (flipCount > flipCounter) {
             double randomNum = Math.random();
-            if (randomNum >= 0.5) {
-                flipResult = "heads";
+            if(randomNum >= 0.5 && flipResult == "heads") {
+                // flipResult = "heads";
                 headsCounter++;
                 System.out.println(flipResult);
+                flipSummary = "It took " + flipCount + " flips " + "to flip " + headsCounter + " head in a row.";
+            }else if (randomNum >= 0.5) {
+                flipResult = "heads";
+                System.out.println(flipResult);
             } else {
                 flipResult = "tails";
                 System.out.println(flipResult);
```

Comment:
Need to reset the headsCounter to 1 if you get a tails. Otherwise it will count total heads, not heads in a row.